pkg/tfgen/generate.go

```diff
@@ -1943,6 +1949,44 @@ func emitFile(fs afero.Fs, relPath string, contents []byte) error {
 	return err
 }
 
+func getUniqueLeafDocsDescriptions(arguments map[docsPath]*argumentDocs, path docsPath) string {
+	leaf := path.leaf()
+
+	uniqueLeaves := getUniqueLeafPaths(arguments)
+	if _, exist := uniqueLeaves[leaf]; exist {
+		for argKey, argDoc := range arguments {
+			if argKey.leaf() == leaf {
+				return argDoc.description
+			}
+		}
+	}
+	return ""
+}
+func getUniqueLeafPaths(arguments map[docsPath]*argumentDocs) map[string]bool {
+	// Read all leaves into a map.
+	//The key is the leaf name, and the value is the number of times that leaf has appeared.
+	leaves := make(map[string]int)
+	for arg := range arguments {
+		_, exist := leaves[arg.leaf()]
+		// if the key already exists, then increase the count
+		if exist {
+			leaves[arg.leaf()]++
+			continue
+		}
+		leaves[arg.leaf()] = 1
+	}
+
+	// Only return leaves with a count of exactly 1.
+	// For quick lookup, this is a map where the key is our leaf, and the value a boolean, which we disregard.
+	uniqueLeavesMap := make(map[string]bool)
```

Comment:
If we disregard the boolean, then we should return a simpler type.
```suggestion
	// For quick lookup, this is a map where the key is our leaf, and the value a boolean, which we disregard.
	uniqueLeavesMap := make(map[string]struct{})
```